packages/components/src/internal/components/user/APIKeysPanel.tsx

```diff
@@ -111,30 +124,55 @@ export const KeyGenerator: FC<KeyGeneratorProps> = props => {
         document.execCommand('copy');
     }, [keyValue]);
 
+    const changeDescription = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+        setDescription(event.target.value);
+    }, []);
+
     return (
-        <>
-            <div className="top-spacing form-group">
-                <button className="btn btn-success api-key__button" onClick={onGenerateKey} disabled={!!keyValue}>
-                    Generate {noun}
-                </button>
-
-                <input
-                    disabled
-                    type="text"
-                    className="form-control api-key__input"
-                    name={type + '_token'}
-                    value={keyValue}
-                />
-                <button
-                    className="btn btn-default api-key__button"
-                    title="Copy to clipboard"
-                    name={'copy_' + type + '_token'}
-                    onClick={onCopyKey}
-                    disabled={!keyValue}
-                >
-                    <i className="fa fa-clipboard"></i>
-                </button>
-            </div>
+        <Modal
+            title={noun}
+            cancelText={keyValue ? 'Done' : undefined}
+            onCancel={onClose}
+            canConfirm={!keyValue}
+            confirmText={type === 'apikey' ? 'Generate ' + noun : undefined}
+            onConfirm={!keyValue && type === 'apikey' ? onGenerateKey : undefined}
+        >
+            {type === 'apikey' && !keyValue && (
+                <div>
+                    <label htmlFor="keyDescription" className="right-spacing">
+                        Description (optional)
```

Comment:
we usually indicate the opposite for field labels (i.e. use asterisks for required and nothing for optional). minor but this seems like it could drop the "(optional)" part of the label.

Reply:
True. I can probably also remove that "right-spacing" class. Left over from previous placement of this element.